controller/common/common.go

```diff
@@ -64,19 +64,36 @@ func (m ChildMap) FindGroupKindName(apiGroup, kind, name string) *unstructured.U
 	return nil
 }
 
-func (m ChildMap) ReplaceChild(child *unstructured.Unstructured) {
+func (m ChildMap) ReplaceChild(prefix string, child *unstructured.Unstructured) {
 	key := childMapKey(child.GetAPIVersion(), child.GetKind())
 	children := m[key]
 	if children == nil {
 		// We only want to replace if it already exists, so do nothing.
 		return
 	}
 	name := child.GetName()
-	if _, found := children[name]; found {
-		children[name] = child
+	if _, found := children[prefix + name]; found {
+		children[prefix + name] = child
 	}
 }
 
+func MakeClusteredChildMap(list []*unstructured.Unstructured) ChildMap {
```

Comment:
nit: Can you add a function comment to explain the difference between this and `MakeChildMap`? The rule is, "call this one if the parent is cluster-scoped," right?

Reply:
I created a new function mostly for the sake of getting something working. Yeah you're guess at the rule is correct.  I wish it was a little simpler for the caller to use, I was thinking about teaching the existing function how to handle the case when the parent is clustered but then we would essentially have a Boolean flag argument, but is usually not a great API.

The other option would be the always use the namespace prefix in the internal client map and strip the prefix when sending the client map to the sync hook. The reason to do it this way would be the hope that treating the client map consistently internally will make it easier to work with and we'd have less issue with it long term.

Reply:
I had another thought: how about adding `parent metav1.Object` as the first argument to `MakeChildMap()`, `Insert()`, `ReplaceChild()`, and so on? It makes sense to me that the format of the child map might depend on something about the parent. By requiring callers to give us the actual parent, we can encapsulate whatever changes we need that depend on the parent. In this case, we know to add a prefix if `parent.GetNamespace() != child.GetNamespace()`.

Reply:
I like that a lot better than adding a `namespaced bool` flag. I'll give that a shot this morning.